simplelink_cc26x2_sdk_2_20_00_36/docs/thread/html/_sources/tirtos/config.rst.txt
Kernel Configuration
--------------------

A TI-RTOS application configures the TI-RTOS kernel using a configuration
(``.cfg`` file) that is found within the project. In :term:`IAR` and
:term:`CCS` projects, this file is found in the application project workspace
under the ``TOOLS`` folder.

The configuration is accomplished by selectively including or *using*
:term:`RTSC` modules available to the kernel. To *use* a module, the ``.cfg``
calls ``xdc.useModule()`` after which it can set various options as defined in
the `TI-RTOS Kernel User Guide`_.

Some of the option that can be configured in the ``.cfg`` file include but are
not limited to:

- Boot options

- Number of Hwi, Swi, and Task priorities

- Exception and Error handling

- The duration of a System tick (the most fundamental *unit* of time in the
  TI-RTOS kernel).

- Defining the application's entry point and interrupt vector

- TI-RTOS heaps and stacks

- Including pre-compiled kernel and TI-RTOS driver libraries

- System providers (for ``System_printf()``)

Whenever a change in the ``.cfg`` file is made, you will rerun the XDCTools'
``configuro`` tool. This step is already handled for you as a pre-build step in
the provided :term:`IAR` and :term:`CCS` examples.

For the |DEVICE|, a TI-RTOS kernel exists in :term:`ROM`. Typically for flash
footprint savings, the ``.cfg`` will include the kernel's :term:`ROM` module as
shown in :numref:`rom-listing`

.. _rom-listing:
.. code-block:: js
    :caption: How to include the TI-RTOS kernel in ROM
    :emphasize-lines: 5,7,10

    /* ================ ROM configuration ================ */
    /*
     * To use BIOS in flash, comment out the code block below.
     */
    var ROM = xdc.useModule('ti.sysbios.rom.ROM');
    if (Program.cpu.deviceName.match(/CC26X2/)) {
        ROM.romName = ROM.CC26X2;
    }
    else if (Program.cpu.deviceName.match(/CC13X2/)) {
        ROM.romName = ROM.CC13X2;
    }

The TI-RTOS kernel in :term:`ROM` is optimized for performance. If additional
instrumentation is required in your application (typically for debugging), you
must include the TI-RTOS kernel in flash which will increase flash memory
consumption. Shown below is a short list of requirements to use the TI-RTOS
kernel in :term:`ROM`.

- ``BIOS.assertsEnabled`` **must** be set to ``false``

- ``BIOS.logsEnabled`` **must** be set to ``false``

- ``BIOS.taskEnabled`` **must** be set to ``true``

- ``BIOS.swiEnabled`` **must** be set to ``true``

- ``BIOS.runtimeCreatesEnabled`` **must** be set to ``true``

- BIOS **must** use the ``ti.sysbios.gates.GateMutex`` module

- ``Clock.tickSource`` **must** be set to ``Clock.TickSource_TIMER``

- ``Semaphore.supportsPriority`` **must** be ``false``

- Swi, Task, and Hwi hooks are **not** permitted

- Swi, Task, and Hwi name instances are **not** permitted

- Task stack checking is **disabled**

- ``Hwi.disablePriority`` **must** be set to ``0x20``

- ``Hwi.dispatcherAutoNestingSupport`` **must** be set to true

For additional documentation in regards to the list described above, see the
`TI-RTOS Kernel User Guide`_.

.. _create_vs_construct:

Creating vs. Constructing
-------------------------
Most TI-RTOS modules commonly have ``_create()`` and ``_construct()`` APIs to
initialize primitive instances. The main **runtime** differences between the
two APIs are memory allocation and error handling.

**Create** APIs perform a memory allocation from the default TI-RTOS heap
before initialization. As a result, the application *must* check the return
value for a valid handle before continuing.

.. _create-example-listing:
.. code-block:: c
    :linenos:
    :caption: Creating a Semaphore
    :emphasize-lines: 5, 7

    Semaphore_Handle sem;
    Semaphore_Params semParams;

    Semaphore_Params_init(&semParams);
    sem = Semaphore_create(0, &semParams, NULL); /* Memory allocated in here */

    if (sem == NULL) /* Check if the handle is valid */
    {
        System_abort("Semaphore could not be created");
    }

**Construct** APIs are given a data structure with which to store the
instance's variables. As the memory has been pre-allocated for the instance,
error checking may not be required after constructing.

.. _construct-example-listing:
.. code-block:: c
    :linenos:
    :caption: Constructing a Semaphore
    :emphasize-lines: 3, 6

    Semaphore_Handle sem;
    Semaphore_Params semParams;
    Semaphore_Struct structSem; /* Memory allocated at build time */

    Semaphore_Params_init(&semParams);
    Semaphore_construct(&structSem, 0, &semParams);

    /* It's optional to store the handle */
    sem = Semaphore_handle(&structSem);


simplelink_cc26x2_sdk_2_20_00_36/docs/ti154stack/html/_sources/ti154stack/configuring-stack.rst.txt
.. _sec-configuring-stack:

Configuring Stack
=================

Mode Configuration
------------------

The TI 15.4-Stack offers three modes of network operations that follow (and as
discussed in this chapter):

-   Beacon mode

-   Nonbeacon mode

-   Frequency-hopping mode

-   BLE Advertising Mode

The ``features.h`` file allows developers to compile-in or compile-out different
15.4-Stack features for different applications. TI 15.4-Stack allows support for
all three modes or allows user to select just one desired mode of network
operation. TI 15.4-Stack can be configured in four different modes of operation
using the ``features.h`` file. Depending on the mode selection, considerable
savings in the executable-image space can be achieved.
:numref:`tab-collector-flags` and :numref:`tab-sensor-flags` provide a summary
of Flash and RAM use of the out-of-box Collector and Sensor Example Application
with different compile options enabled.

#.  ``FEATURE_ALL_MODES``: When this compile flag is defined, the image is
    compiled with all the three modes of operation (frequency-hopping mode,
    beacon-enabled mode and non-beacon mode) and the configuration file
    ``config.h`` can be used to select the specific mode for network operation.
    This feature allows flexibility to select any mode for the device. For the
    out-of-box collector and sensor example application, this feature is
    enabled.::

        /*! If defined, builds the image with all the modes of operation (frequency hopping, beacon mode and non beacon mode) */
        #define FEATURE_ALL_MODES

#.  ``FEATURE_FREQ_HOP_MODE``: Defining this compile flag will compile only the
    frequency hopping mode of operation in the final executable image. For out
    of box example application, you would need to disable the compile option
    ``FEATURE_ALL_MODES`` and then enable this compile option as in the
    following:::

        /*! If defined, builds the image with all the modes of operation (frequency hopping, beacon mode and non beacon mode) */
        #undef FEATURE_ALL_MODES

        /*! If defined, builds the image with the frequency mode of operation */
        #define FEATURE_FREQ_HOP_MODE

#.  ``FEATURE_BEACON_MODE``: Defining this compile flag will compile only the
    beacon mode of operation in the final executable image. For out of box
    example application, you would need to disable the compile option
    ``FEATURE_ALL_MODES`` and then enable this compile option as in the
    following:::

        /*! If defined, builds the image with all the modes of operation (frequency hopping, beacon mode and non beacon mode) */
        #undef FEATURE_ALL_MODES

        /*! If defined, builds the image with beacon mode of operation */
        #define FEATURE_BEACON_MODE

#.  ``FEATURE_NON_BEACON_MODE``: Defining this compile flag will compile only
    the non-beacon mode of operation in the final executable image.::

        /*! If defined, builds the image with all the modes of operation (frequency hopping, beacon mode and non beacon mode) */
        #undef FEATURE_ALL_MODES

        /*! If defined, builds the image with non beacon mode of operation */
        #define FEATURE_NON_BEACON_MODE

#.  FEATURE_BLE_SUPPORT: Defining this compile flag will enable BLE operations
    in the stack. CONFIG_BLE_SUPPORT will also need to be configured in config.h for
    the example application to send out BLE beacons.

In addition to the compile flags listed previously, the
``FEATURE_FULL_FUNCTION_DEVICE`` compile flag is required for the PAN
Coordinator device (see the out-of-box Collector Example Application) to perform
the role as a central node in the network.

Also, the ``FEATURE_MAC_SECURITY`` compile flag is added in the ``features.h``
file to allow the ability to turn the MAC layer security on and turn off in the
compile executable image. If the mac layer security is turned off, you will need
the version of the stack library with no MAC security.

To build the image with no security, perform the steps that follow:

#.  Select the linker *File Search Path* option.

#.  Modify to include the ``maclib_nosecure.a`` instead of ``maclib_secure.a``
    library file (shown in :numref:`fig-stack-library`).

    .. figure:: resources/fig-stack-library.jpeg
        :name: fig-stack-library
        :align: center

        Changing TI 15.4 Stack library

The ``PREAMBLE_COMPATIBILITY`` compile flag allows the stack to be compatible
with previous versions of TI 15.4-Stack. When this flag is not defined the stack
uses a IEEE 802.15.4g compliant preamble. If this flag is enabled, a new version
of the stack library is required.

Similar to the steps above, to change the stack library for compatibility:

#.  Select the linker *File Search Path* option.
#.  Modify to include the ``maclib_nosecureLegacy.a`` or
    ``maclib_secureLegacy.a`` library file depending on whether or not security
    is enabled or disabled.

.. table:: Out-of-Box Collector Example Application Flash and RAM Usage Summary
    With Various Compile-Option Combinations
    :name: tab-collector-flags

    +----------------------------+----------------+--------------+
    | **Compile Option Enabled** | **FLASH (KB)** | **RAM (KB)** |
    +============================+================+==============+
    | FEATURE_ALL_MODES          | 103            | 15           |
    +----------------------------+----------------+--------------+
    | FEATURE_FREQ_HOP_MODE      | 98.1           | 14.6         |
    +----------------------------+----------------+--------------+
    | FEATURE_NON_BEACON_MODE    | 84.6           | 13.9         |
    +----------------------------+----------------+--------------+
    | FEATURE_BEACON_MODE        | 88.8           | 14.2         |
    +----------------------------+----------------+--------------+

.. table:: Out-of-Box Sensor Example Application Flash and RAM Usage Summary
    With Various Compile-Option Combinations
    :name: tab-sensor-flags

    +----------------------------+----------------+--------------+
    | **Compile Option Enabled** | **FLASH (KB)** | **RAM (KB)** |
    +============================+================+==============+
    | FEATURE_ALL_MODES          | 103            | 13           |
    +----------------------------+----------------+--------------+
    | FEATURE_FREQ_HOP_MODE      | 104.3          | 12.7         |
    +----------------------------+----------------+--------------+
    | FEATURE_NON_BEACON_MODE    | 89.6           | 12.1         |
    +----------------------------+----------------+--------------+
    | FEATURE_BEACON_MODE        | 92.1           | 12.1         |
    +----------------------------+----------------+--------------+

.. note:: ``FEATURE_BLE_SUPPORT`` will add around 6KB of flash and 1.2KB of RAM
    to any of the modes described above except for ``FEATURE_ALL_MODES``.

CC1190 PA/LNA Support
---------------------

CC1190 PA/LNA can be used to enhance the transmit power of the device using the
external Power Amplifier. If the system uses an external CC1190 external PA/LNA,
the stack should be configured to use the appropriate power tables. This can be
achieved by setting the PIB attribute ``ApiMac_attribute_rangeExtender`` to 1.
It is required to set this attribute first before setting the PIB attribute
``ApiMac_attribute_phyTransmitPowerSigned`` to set the required power level.
When CC1190 is enable the following output power values will only be supported:
18, 23, 25, 26 and 27 dbm. When other values are requested, a status of
ApiMac_status_invalidParameter will be returned. For the :term:`OOB` |TIMAC|
applications, CC1190 PA/LNA support can be enabled by setting the configuration
option ``CONFIG_RANG_EXT_MODE`` to ``APIMAC_HIGH_GAIN_MODE``.

CC1352P1 and CC1352P2 Support
-----------------------------

The CC1352P1 and CC1352P2 have an integrated power amplifier that can be used to
increase transmit power of the device. 

The CC1352P1 device adds a power amplifier to the Sub1-GHz RF path. This will
increase the maximum transmit power of the device from +14 dBm to +20 dBm. For
the :term:`OOB` |TIMAC| applications, power amplifier support can be enabled by
setting the configuration option ``CONFIG_PA_TYPE`` to ``APIMAC_HIGH_PA``.

The CC1352P2 device adds a power amplifier to the 2.4 GHz RF path. This will
increase the maximum transmit power from +5 dBm to +20 dBm. For the :term:`OOB`
|TIMAC| applications, power amplifier support can be enabled by setting the
configuration option ``CONFIG_PA_TYPE`` to ``APIMAC_HIGH_PA``.

.. note::
    The CC1352P2 will only increase transmit power for projects configured for
    operation in 2.4 GHz. The symbol ``FREQ_2_4G`` should be defined in the
    projects predefined symbols for operation in 2.4 GHz.

Bluetooth Low Energy Advertiser
-------------------------------

The TI 15.4-Stack supports Bluetooth Low Energy (BLE) advertisements,
specifically, non-connectable undirected advertisements in simultaneous
operation with the TI 15.4 stack. Such simultaneous operation is available only
on the dual-band devices such as the CC1350.
The TI 15.4 stack uses uBLE (pronounced micro BLE) stack subsystem which is a
tiny light-weight non-connection based protocol stack module that supports only
BLE Broadcaster role. It does not operate using a separate TI-RTOS stack but is
integrated in the application. This saves memory and overhead required to
maintain a separate RTOS task.  Since uBLE supports only BLE Broadcaster role,
there is no strict timing constraint. Therefore uBLE radio operation have lower
priority compared to 15.4 radio operations. This implies that 15.4 radio
operations will preempt uBLE operations.

.. note:: The BLE advertisement payload is fully modifiable. Any of the
    out-of-box example sensor applications can be configured to enable BLE
    advertisements.

Configuring uBLE Stack
^^^^^^^^^^^^^^^^^^^^^^

``FEATURE_BLE``: Defining this compile flag in ``features.h`` will compile the
BLE advertisement feature.::

    /*! If defined, builds the image wih BLE Advertiser enabled */
    #define FEATURE_BLE

``CONFIG_BLE_SUPPORT``: Apart from ``FEATURE_BLE`` this configuration must be
set to true in ``config.h``:::

    /*! Enable BLE advertiser   */
    #define CONFIG_BLE_SUPPORT          true

The BLE advertisement interval can be modified as follows in ``ble_adv.c``:::

    #define DEFAULT_ADVERTISING_INTERVAL    2000

.. note:: The units are in terms of 0.625 milliseconds. Hence setting it to
    2000 will result in BLE advertisements being sent out every 1.25 seconds.

.. _sec-security:

Security
--------

The TI 15.4-Stack supports AES encryption as defined by the IEEE 802.15.4
Specification. The application is responsible for management of the keys. The
out-of-box example application of the TI 15.4-Stack demonstrates how to use
security with the TI 15.4-Stack.

.. note:: TI 15.4-Stack supports pair wise keys.

simplelink_cc26x2_sdk_2_20_00_36/examples/rtos/CC26X2R1_LAUNCHXL/ti154stack/sensor/2_4g/config.h
/******************************************************************************

 @file config.h

 @brief TI-15.4 Stack configuration parameters for Sensor applications

 Group: WCS LPC
 Target Device: CC2652

 ******************************************************************************
 
 Copyright (c) 2016-2018, Texas Instruments Incorporated
 All rights reserved.

 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions
 are met:

 *  Redistributions of source code must retain the above copyright
    notice, this list of conditions and the following disclaimer.

 *  Redistributions in binary form must reproduce the above copyright
    notice, this list of conditions and the following disclaimer in the
    documentation and/or other materials provided with the distribution.

 *  Neither the name of Texas Instruments Incorporated nor the names of
    its contributors may be used to endorse or promote products derived
    from this software without specific prior written permission.

 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
 OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
 WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
 OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

 ******************************************************************************
 Release Name: simplelink_cc26x2_sdk_2_20_00_36
 Release Date: 2018-06-27 10:07:01
 *****************************************************************************/
#ifndef CONFIG_H
#define CONFIG_H

/******************************************************************************
 Includes
 *****************************************************************************/
#include "api_mac.h"

#ifdef __cplusplus
extern "C"
{
#endif

/******************************************************************************
 Constants and definitions
 *****************************************************************************/
/* config parameters */
/*! Security Enable - set to true to turn on security */
#define CONFIG_SECURE                true
/*! PAN ID */
#define CONFIG_PAN_ID                0xFFFF
/*! FH disabled as default */
#define CONFIG_FH_ENABLE             false

/*!
 Beacon order, value of 15 indicates non beacon mode,
 8 is a good value for beacon mode
*/
#define CONFIG_BEACON_ORDER          15
/*!
 Superframe order, value of 15 indicates non beacon mode,
 8 is a good value for beacon mode
*/
#define CONFIG_SUPERFRAME_ORDER      15
/*! Maximum number of message failure, to indicate sync loss */
#define CONFIG_MAX_DATA_FAILURES     3
/*!
 Maximum number of attempts for association in FH mode
 after reception of a PAN Config frame
 */
#define CONFIG_FH_MAX_ASSOCIATION_ATTEMPTS    3
/* Interval for scan backoff */
#define CONFIG_SCAN_BACKOFF_INTERVAL  5000
/* Interval for delay between orphan notifications */
#define CONFIG_ORPHAN_BACKOFF_INTERVAL 300000

/*! Setting for Phy ID */
#define CONFIG_PHY_ID                (APIMAC_PHY_ID_NONE)

/*! Setting for channel page */
#define CONFIG_CHANNEL_PAGE          (APIMAC_CHANNEL_PAGE_NONE)

/*! scan duration */
#define CONFIG_SCAN_DURATION         5

/*!
 Coordinator Short Address When Operating with FH Enabled.
 */
#define FH_COORD_SHORT_ADDR 0xAABB
/*!
 Range Extender Mode setting.
 The following modes are available.
 APIMAC_NO_EXTENDER - does not have PA/LNA
 APIMAC_HIGH_GAIN_MODE - high gain mode
 To enable CC1190, use
 #define CONFIG_RANGE_EXT_MODE       APIMAC_HIGH_GAIN_MODE
*/
#define CONFIG_RANGE_EXT_MODE       APIMAC_NO_EXTENDER

/*!
 High PA Mode setting.
 The following modes are available.
 APIMAC_DEFAULT_PA - does not have High PA
 APIMAC_HIGH_PA - High PA
 To enable PA, use
 #define CONFIG_PA_TYPE       APIMAC_HIGH_PA
*/
#define CONFIG_PA_TYPE       APIMAC_DEFAULT_PA

/*! Setting Default Key*/
#define KEY_TABLE_DEFAULT_KEY {0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0,\
                               0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}

/*!
 Channel mask used when CONFIG_FH_ENABLE is false.
 Each bit indicates if the corresponding channel is to be scanned
 First byte represents channels 0 to 7 and the last byte represents
 channels 128 to 135.
 For byte zero in the bit mask, LSB representing Ch0.
 For byte 1, LSB represents Ch8 and so on.
 e.g., 0x01 0x10 represents Ch0 and Ch12 are included.
 The default of 0x0F represents channels 0-3 are selected.
 APIMAC_STD_US_915_PHY_1 (50kbps/2-FSK/915MHz band) has channels 0 - 128.
 APIMAC_STD_ETSI_863_PHY_3 (50kbps/2-FSK/863MHz band) has channels 0 - 33.
 APIMAC_GENERIC_CHINA_433_PHY_128 (50kbps/2-FSK/433MHz band) has channels 0 - 6.
 IEEE 802.15.4 has channels 11 - 26.
*/
#define CONFIG_CHANNEL_MASK           { 0x00, 0x78, 0x00, 0x00, 0x00, 0x00, \
                                        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
                                        0x00, 0x00, 0x00, 0x00, 0x00 }
/*!
 Channel mask used when CONFIG_FH_ENABLE is true.
 Represents the list of channels on which the device can hop.
 When CONFIG_RX_ON_IDLE is true, the actual sequence will
 be based on DH1CF function. When it is set to false, the sequence
 shall be a linear hopping over available channels in ascending order and
 shall be used to change channel during the join phase.
 It is represented as a bit string with LSB representing Ch0.
 e.g., 0x01 0x10 represents Ch0 and Ch12 are included.
 */
#define CONFIG_FH_CHANNEL_MASK        { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, \
                                        0xFF, 0x00, 0x00, 0x00, 0x00, 0x00, \
                                        0x00, 0x00, 0x00, 0x00, 0x00,}
/* FH related config variables */
/*!
 List of channels to target the Async frames
 It is represented as a bit string with LSB representing Ch0
 e.g., 0x01 0x10 represents Ch0 and Ch12 are included
 It should cover all channels that could be used by a target device in its
 hopping sequence. Channels marked beyond number of channels supported by
 PHY Config will be excluded by stack. To avoid interference on a channel,
 it should be removed from Async Mask and added to exclude channels
 (CONFIG_CHANNEL_MASK).
 */
#define FH_ASYNC_CHANNEL_MASK         { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, \
                                        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, \
                                        0xFF, 0xFF, 0xFF, 0xFF, 0xFF }

/*! Rx on when idle, false for sleepy device, true for non sleepy device */
#define CONFIG_RX_ON_IDLE          false

/* FH related config variables */
/*!
 The number of non sleepy channel hopping end devices to be supported.
 It is to be noted that the total number of non sleepy devices supported
  must be less than 50. Stack will allocate memory proportional
 to the number of end devices requested.
 */
#define FH_NUM_NON_SLEEPY_HOPPING_NEIGHBORS  2
/*!
 The number of non sleepy fixed channel end devices to be supported.
 It is to be noted that the total number of non sleepy devices supported
  must be less than 50. Stack will allocate memory proportional
 to the number of end devices requested.
 */
#define FH_NUM_NON_SLEEPY_FIXED_CHANNEL_NEIGHBORS  2
/*!
 Dwell Time: The duration for which a non sleepy end device shall
 stay on a specific channel before hopping to next channel.
 */
#define CONFIG_DWELL_TIME            250

/*! MAC Parameter */
/*! Min BE - Minimum Backoff Exponent */
#define CONFIG_MIN_BE   3
/*! Max BE - Maximum Backoff Exponent */
#define CONFIG_MAX_BE   5
/*! MAC MAX CSMA Backoffs */
#define CONFIG_MAC_MAX_CSMA_BACKOFFS   4
/*! macMaxFrameRetries - Maximum Frame Retries */
#define CONFIG_MAX_RETRIES   3

/*! Polling interval in milliseconds */
#define CONFIG_POLLING_INTERVAL      500
/*! PAN Advertisement Solicit trickle timer duration in milliseconds */
#define CONFIG_PAN_ADVERT_SOLICIT_CLK_DURATION    6000
/*! PAN Config Solicit trickle timer duration in milliseconds */
#define CONFIG_PAN_CONFIG_SOLICIT_CLK_DURATION    6000
/*! Default Reporting Interval - in milliseconds */
#define CONFIG_REPORTING_INTERVAL  2000

/*! FH Poll/Sensor msg start time randomization window */
#define CONFIG_FH_START_POLL_DATA_RAND_WINDOW   10000

/*! If enabled, the periodic sensor message shall be sent as a fixed size
 * packet of specified size. If set to 0, the periodic sensor message shall be
 * of type sensor data specified in smsgs.h
 */
#define SENSOR_TEST_RAMP_DATA_SIZE   0

/*! value for ApiMac_FHAttribute_netName */
#define CONFIG_FH_NETNAME            {"FHTest"}
/*!
 Value for Transmit Power in dBm
 For US and ETSI band, Default value is 10, allowed values are
 -10, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 and 14dBm.
 For China band, allowed values are 6, 10, 13, 14 and 15dBm.
 For CC1190, allowed values are between 18, 23, 25, 26 and 27dBm.
 When the nodes in the network are close to each other
 lowering this value will help reduce saturation */

#if CONFIG_PA_TYPE
#define CONFIG_TRANSMIT_POWER        20
#else
#define CONFIG_TRANSMIT_POWER        0
#endif

/*!
* Enable this mode for certfication.
* For FH certification, CONFIG_FH_ENABLE should
* also be enabled
*/
#define CERTIFICATION_TEST_MODE     false

#ifdef POWER_MEAS
/*!
 Power profile to be used when Power MEAS is enabled.
 Profile 1 - POLL_ACK - Polling Only
 Profile 2 - DATA_ACK - 20 byte application data + ACK from sensor to collector
 Profile 3 - POLL_DATA - Poll + received Data from collector
 Profile 4 - SLEEP - No Poll or Data. In Beacon mode, beacon RX would occur
 */
#define POWER_TEST_PROFILE  DATA_ACK
#endif

/* Check if stack level security is enabled if application security is enabled */
#if CONFIG_SECURE
#if !defined(FEATURE_MAC_SECURITY)
#error "Define FEATURE_MAC_SECURITY or FEATURE_ALL_MODES in features.h to \
        be able to use security at application level"
#endif
#endif

#ifdef __cplusplus
}
#endif

#endif /* CONFIG_H */

